### ISSUE TITLE
Session affinity may be preferable for stateful workloads hosted in Azure Container Apps

### ISSUE THREAD

**Issue by BenjaminEngeset** [Azure/PSRule.Rules.Azure#2188]
### Suggested rule

Azure Container Apps now supports session affinity, also known as sticky sessions.

This feature is in public preview.

Session affinity enables you to route all requests from a single client to the same Container Apps replica. This is useful for stateful workloads that require session affinity.

Session affinity is only supported when the app is in [single revision mode](https://learn.microsoft.com/nb-no/azure/container-apps/revisions#single-revision-mode) and the ingress type is HTTP.

Default behaviour should be off as stateless is the general recommendation, but customers can modify this by configuration in PSRule by enabling the rule.

### Pillar

Performance Efficiency

### Additional context

https://learn.microsoft.com/nb-no/azure/container-apps/sticky-sessions
https://learn.microsoft.com/en-us/azure/templates/microsoft.app/containerapps#ingress

**Comment by BernieWhite** [comment]
@BenjaminEngeset I feel this rule should be actually to disable affinity instead of enable it as the WAF generally discourages affinity.

**Comment by BernieWhite** [comment]
@BenjaminEngeset Correct. I think you suggestion of _Performance Efficiency_ is correct because the general side effect that client affinity creates is unbalanced distribution of load. i.e. one instance is significantly more loaded then other instances, and the loaded instance performs poorly compared with other instances.

**Comment by BenjaminEngeset** [comment]
> @BenjaminEngeset Correct. I think you suggestion of _Performance Efficiency_ is correct because the general side effect that client affinity creates is unbalanced distribution of load. i.e. one instance is significantly more loaded then other instances, and the loaded instance performs poorly compared with other instances.

@BernieWhite Great! Thanks, then I know what to do.
